### PR TITLE
Jvenstad/convert more deployment tests

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Ah, forgot I had an open PR. Pushed two more commits.

**Comment by jonmv** [comment]
Urgh, ConcurrentModificationException in ConfigServerApiHandlerTest — fixed.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Controller.java [hunk 1]
```diff
@@ -78,28 +78,26 @@
      */
     @Inject
     public Controller(CuratorDb curator, RotationsConfig rotationsConfig,
-                      ZoneRegistry zoneRegistry,
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Controller.java]: `ZoneRegistryMock` can be removed from `ControllerContainerTest` now.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java [hunk 2]
```diff
@@ -1,153 +0,0 @@
...
-
-import static com.yahoo.vespa.hosted.controller.api.integration.deployment.JobType.component;
-import static org.junit.Assert.assertFalse;
-
-/**
- * Provides testing of controller functionality accessed through the container
- * 
- * @author bratseth
- */
-public class ContainerControllerTester {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java]: Great, I could never remember the difference between this and `ControllerContainerTest`.

**Review by mpolden — APPROVED** [review]
💯

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Controller.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Controller.java]: Indeed.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java]: :D

**Pull request merged**